ethereum_history_api/oracles/src/noir/oracles/server/handlers.ts

```diff
@@ -49,3 +51,13 @@ export async function getProofHandler(params: ForeignCallParams, { client }: Ser
   const result = encodeForeignCallResult(noirOutputs);
   return result;
 }
+
+export async function getReceiptHandler(
+  params: ForeignCallParams,
+  { client }: ServerParams
+): Promise<ForeignCallResult> {
+  const noirArguments = decodeNoirArguments(params);
+  const noirOutputs = await getReceiptOracle(client, noirArguments);
+  const result = encodeForeignCallResult(noirOutputs);
+  return result;
+}
```

Comment:
Idea: extract function: `getOracleHandler = (oracle) => (client, noirArguments) `

Reply:
rename: `decodeNoirArguments` => `decodeOracleJsonRpcArguments` (?)

Reply:
encodeForeignCallResult => ?

Reply:
I don't see the reason why we should sometimes name things encode/decode and sometimes wrap/unwrap.
This code lives in server context, so here encoding/decoding has server meaning.
That code lives in oracle context, so there it has a different meaning.